Client/Application/LeanplumIntegration.swift

```diff
@@ -0,0 +1,196 @@
+            return
+        }
+
+        switch settings.environment {
+        case .development:
+            Logger.browserLogger.info("LeanplumIntegration - Setting up for Development")
+            Leanplum.setDeviceId(ASIdentifierManager.shared().advertisingIdentifier.uuidString)
+            Leanplum.setAppId(settings.appId, withDevelopmentKey: settings.key)
+        case .production:
+            Logger.browserLogger.info("LeanplumIntegration - Setting up for Production")
+            Leanplum.setAppId(settings.appId, withProductionKey: settings.key)
+        }
+        Leanplum.syncResourcesAsync(true)
+        setupTemplateDictionary()
+
+        var userAttributesDict = [AnyHashable: Any]()
+        userAttributesDict["Focus Installed"] = "false"
+        userAttributesDict["Klar Installed"] = "false"
+        userAttributesDict["Alternate Mail Client Installed"] = "mailto:"
+
+        if let focusURL = URL(string: "firefox-focus://"), UIApplication.shared.canOpenURL(focusURL) {
+            userAttributesDict["Focus Installed"] = "true"
+        }
+
+        if let klarURL = URL(string: "firefox-klar://"), UIApplication.shared.canOpenURL(klarURL) {
+            userAttributesDict["Klar Installed"] = "true"
+        }
+
+        if let mailtoScheme = self.profile?.prefs.stringForKey(PrefsKeys.KeyMailToOption), mailtoScheme != "mailto:" {
+            userAttributesDict["Alternate Mail Client Installed"] = mailtoScheme
+        }
+
+        Leanplum.start(userAttributes: userAttributesDict)
+
+        Leanplum.track(LeanplumEventName.openedApp.rawValue)
+    }
+
+    // Events
+
+    func track(eventName: LeanplumEventName) {
+        if shouldSendToLP() {
+            Leanplum.track(eventName.rawValue)
+        }
+    }
+
+    func track(eventName: LeanplumEventName, withParameters parameters: [String: AnyObject]) {
+        if shouldSendToLP() {
+            Leanplum.track(eventName.rawValue, withParameters: parameters)
+        }
+    }
+
+    // States
+
+    func advanceTo(state: String) {
+        if shouldSendToLP() {
+            Leanplum.advance(to: state)
+        }
+    }
+
+    // Data Modeling
+
+    func setupTemplateDictionary() {
+        if shouldSendToLP() {
+            LPVar.define("Template Dictionary", with: ["Template Text": "", "Button Text": "", "Deep Link": "", "Hex Color String": ""])
+        }
+    }
+
+    func getTemplateDictionary() -> [String:String]? {
+        if shouldSendToLP() {
+            return Leanplum.object(forKeyPathComponents: ["Template Dictionary"]) as? [String : String]
+        }
+        return nil
+    }
+
+    func getBoolVariableFromServer(key: String) -> Bool? {
+        if shouldSendToLP() {
+            return Leanplum.object(forKeyPathComponents: [key]) as? Bool
+        }
+        return nil
+    }
+
+    // Utils
+    
+    func setEnabled(_ enabled: Bool) {
+        self.enabled = enabled
+        if self.enabled {
+            self.start()
+        }
```

Comment:
I don't think this is good enough. There is a user expectation that when they turn off *Send Anonymous User Data* that at that point they are opted out from data collection. We make this happens for Telemetry and Adjust, so Leanplum still collecting and sending data back cannot happen after this toggle has flipped to No.

I understand this will take effect the next time the user starts the application, but we have to be strict here. On a modern phone it is possible that Firefox runs for days or even weeks without being actually restarted.

Is there anything in the LeanPlum SDK that we can call to pause or stop the data collection?

(I know we have `shouldSendToLP()` but LP also does its down session tracking outside of calls to `Leanplum.track()`)